community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/fusion/FusionIndexProvider.java

```diff
@@ -145,13 +142,13 @@ private void writeFailure( String indexName, StringBuilder builder, IndexProvide
     @Override
     public InternalIndexState getInitialState( long indexId, SchemaIndexDescriptor descriptor )
     {
-        InternalIndexState[] states = providers.instancesAs( new InternalIndexState[INSTANCE_COUNT], p -> p.getInitialState( indexId, descriptor ) );
-        if ( Arrays.stream( states ).anyMatch( state -> state == FAILED ) )
+        Iterable<InternalIndexState> states = providers.flatMap( p -> p.getInitialState( indexId, descriptor ) );
```

Comment:
you do not need to go over those states using streams, convert those to list and do contains instead